### PR TITLE
refactor: clean up enabling strict mode

### LINKED ISSUE #32: Support typescript strict mode
**What would you like to be added**:

Support strict type checking settings, such as disallowing `implicit any` or missing declarations. Change config properties in `tsconfig.json` to enable these features (i.e. `strict: true`). Make the necessary changes to the codebase to pass checks under these settings.

**Why is this needed**:

Helps further prevent bugs with more thorough type validation throughout the codebase.

**Tasks**:
- [x] `noImplicitAny` @MeisterRed 
- [x] `strictNullChecks` @zthaver
- [x] `strictPropertyInitialization` @zthaver 
- [x] `strictBindCallApply` @pynnl 
- [x] `noImplicitThis` @pynnl 
- [x] `strictFunctionTypes` @pynnl 
- [x] `alwaysStrict` @pynnl 
- [ ] combine strict options in `tsconfig` (should be done after the aboves) @pynnl 
- [x] add type for `jsdom` and `express`, and remove redundancy @pynnl

### LINKED ISSUE #199: Remove Explicit Casting
**What would you like to be changed**: Several files such as Browser.ts and Session.ts use explicit casting, this should use guards and other logic to determine types.

**Why is this needed**: To avoid runtime errors in future.

### PR DESCRIPTION
**Pull request by pynnl**
What this PR does:

- Enable strict mode on temporarily disabled files in #202 
- Fix eslint warning (`explicit any` and `non-null assertion`)
- Remove most explicit casting
- Combine strict options into single `strict: true` to make sure the tests pass
- Fix some minor mistakes

close #32, close #199

### CODE REVIEW

**Review by MeisterRed — APPROVED** [review]

**Review by zthaver — APPROVED** [review]

**Review by manekenpix — APPROVED** [review]

**Review by poftadeh — COMMENTED** [review]
Didn't get to submit this in time, but here's my review just in case. Looks good overall.

File: src/WebElement/WebElement.ts [hunk 1]
```diff
@@ -75,18 +71,20 @@ class WebElement {
...
+    tagName: T,
+  ): HTMLElementTagNameMap[T] | null {
     let { parentElement: nextParent } = this.element;
-
-    const isMatchingOrFalsy = (): boolean =>
-      !nextParent || nextParent.tagName.toLowerCase() === tagName.toLowerCase();
-
-    while (!isMatchingOrFalsy()) {
-      const { parentElement } = nextParent as HTMLElement;
+    document.createElement;
```
[the remark this answers is not in the document]
Comment (pynnl) [reply src/WebElement/WebElement.ts:78]: Nice catch. Thanks.

File: src/WebElement/WebElement.ts [hunk 2]
```diff
@@ -75,18 +71,20 @@ class WebElement {
...
+    while (
+      nextParent &&
+      nextParent.tagName.toLowerCase() !== tagName.toLowerCase()
```
[the remark this answers is not in the document]
Comment (pynnl) [reply src/WebElement/WebElement.ts:81]: I check it seperately in order to have the type guard for `nextParent` inside the loop.

File: src/utils/utils.ts [hunk 3]
```diff
@@ -230,3 +198,13 @@ export const isFrameElement = (
 ): element is HTMLFrameElement => {
   return element.localName === 'frame';
 };
+
+export const isJsonWebElement = (
+  value: unknown,
+): value is { [ELEMENT]: string } =>
+  isObject(value) && isString(value[ELEMENT]);
```
[the remark this answers is not in the document]
Comment (pynnl) [reply src/utils/utils.ts:205]: This function is to check the [web element abstraction ](https://www.w3.org/TR/webdriver1/#elements) from the client, not the instance of `WebElement` class.

**Pull request merged**